stix_shifter/stix_translation/src/modules/guardium/query_constructor.py

```diff
@@ -0,0 +1,523 @@
+from stix_shifter.stix_translation.src.patterns.pattern_objects import ObservationExpression, ComparisonExpression, \
+    ComparisonExpressionOperators, ComparisonComparators, Pattern, \
+    CombinedComparisonExpression, CombinedObservationExpression, ObservationOperators
+from stix_shifter.stix_translation.src.utils.transformers import TimestampToMilliseconds
+from stix_shifter.stix_translation.src.utils.transformers import TimestampToGuardium
+from stix_shifter.stix_translation.src.json_to_stix import observable
+from stix_shifter.stix_translation.src.utils.stix_pattern_parser import parse_stix
+from stix_shifter.stix_translation.src.utils import transformers
+#
+# Following added by Subroto
+import logging
+import re
+import json
+import array
+import datetime
+import re
+import sys
+import copy
+
+# Source and destination reference mapping for ip and mac addresses.
+# Change the keys to match the data source fields. The value array indicates the possible data type that can come into from field.
+#REFERENCE_DATA_TYPES = {"QUERY_FROM_DATE": ["start"],
+#                        "QUERY_TO_DATE": ["end"],"OSUser":["%"],"DBUser":"newuser",
+#                        "SHOW_ALIASES":["TRUE","FALSE"],"REMOTE_SOURCE":["%"]}
+REFERENCE_DATA_TYPES = {}
+
+#logger = logging.getLogger(__name__)
+
+class QueryStringPatternTranslator:
+    # Change comparator values to match with supported data source operators
+    comparator_lookup = {
+        ComparisonExpressionOperators.And: "AND",
+        ComparisonExpressionOperators.Or: "OR",
+#        ComparisonComparators.GreaterThan: ">",
+#        ComparisonComparators.GreaterThanOrEqual: ">=",
+#        ComparisonComparators.LessThan: "<",
+#        ComparisonComparators.LessThanOrEqual: "<=",
+        ComparisonComparators.Equal: "=",
+#        ComparisonComparators.NotEqual: "!=",
+#        ComparisonComparators.Like: "LIKE",
+#        ComparisonComparators.In: "IN",
+#        ComparisonComparators.Matches: 'LIKE',
+        # ComparisonComparators.IsSubSet: '',
+        # ComparisonComparators.IsSuperSet: '',
+        ObservationOperators.Or: 'AND',
+        # Treat AND's as OR's -- Unsure how two ObsExps wouldn't cancel each other out.
+        ObservationOperators.And: 'AND'
+    }
+
+    def __init__(self, pattern: Pattern, data_model_mapper):
+        self.dmm = data_model_mapper
+        self.pattern = pattern
+#
+# Now reportParamsPassed is an json object which is pointing to an array of Json Objects (reportParamsArray)
+        self.reportParamsPassed = {}
+        self.reportParamsArray = []
+        self.reportParamsArraySize = 0
+        self.translated = self.parse_expression(pattern)
+        self.transformers = transformers.get_all_transformers()
+        #Read reference data
```

Comment:
this is not the safe/clean way to read a json file and you would also need to do file not found error handling. you can create a function like i have posted below and pass the relative path of the json file json/reference_data_types4Query.json. you can declare the json file names globally.

    @staticmethod
    def load_json(rel_path_of_file):
        relevance_json_path = path.abspath(path.join(path.join(__file__, ".."), rel_path_of_file))
        if path.exists(relevance_json_path):
            with open(relevance_json_path) as f_obj:
                return json.load(f_obj)
        else:
            raise FileNotFoundError

same goes to the rest of the json file loading statements.